src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            addrs[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            addrs[i].Default_Address__c = true;
+            addrs[i].MailingStreet__c = 'New Default Street' + i;
+            addrs[i].MailingCity__c = 'New Default City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;        
+        Test.stopTest();
+
+        // verify that the Contact addresses are new
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('New Default Street'));
+            system.assert(contact.MailingCity.contains('New Default City'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+        
+        // verify the previous addresses got Default cleared.
+        // and verify latest start date and latest end date appropriately set.
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c, Latest_Start_Date__c, 
+        Latest_End_Date__c from Address__c];
+        system.assertEquals(4, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            boolean fNewDefault = (addr.MailingStreet__c.contains('New Default Street'));
+            system.assertEquals(fNewDefault, addr.Default_Address__c);
+            if (fNewDefault) {
+               system.assertEquals(system.today(), addr.Latest_Start_Date__c);
+               system.assertEquals(null, addr.Latest_End_Date__c);
+            } else {
+               system.assertEquals(system.today(), addr.Latest_End_Date__c);                
+            }
+        }        
+    }
+    
+    /*********************************************************************************************************
+    @description Insert new default addresses to Contact with existing default addresses, and then go back  
+    verify:
+        Contact address matches original default
+        new default addresses no longer marked default
+    **********************************************************************************************************/            
+    @isTest
+    public static void insertNewDefaultAddrAndReset() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+        List<Address__c> ListAddrOriginal = contactsAddrs.addrs;
```

Comment:
Variables should use the `lowerCamelCase` naming convention. Please rename accordingly.